### PR TITLE
Feature/ios track fullscreen controls

### PR DESCRIPTION
**Pull request by CHaNGeTe**
Tracks fullscreen state when changed via controls, and call callbacks if needed

### CODE REVIEW

**Comment by ashnfb** [comment]
@CHaNGeTe  this is very similar to fixes I have been making; although I added an additional bit of code to handle situations with fullscreen rotation.  Let me know if you want to merge your PR, and I can keep my additional code for the future.

https://github.com/nfb-onf/react-native-video/commit/a17b183f105cbd24743447db8598b916b6da842b

**Comment by CHaNGeTe** [comment]
Sorry Ash! Didn't remember you were working on this.

Depending on the ETA of your PR we can release this as quick fix. Or wait for your PR itself. What do you think?

**Comment by ashnfb** [comment]
hi @CHaNGeTe, would you mind if I update this PR with additional changes?

**Comment by CHaNGeTe** [comment]
feel free!

**Review by jensellfors — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "react-native-video",
-    "version": "4.4.3",
+    "version": "4.4.4",
```
Comment (jensellfors) [inline package.json]: This should be 4.4.5 I guess?

**Review by ashnfb — APPROVED** [review]

**Comment by jonathangreco** [comment]
Is this PR add a props to remove fullscreen button on iOS ?

**Comment by ashnfb** [comment]
@jonathangreco it is not.  It's code for optional callbacks when entering or exiting fullscreen

**Comment by luco** [comment]
merge this

**Comment by alimek** [comment]
this fixed problem for me with https://github.com/react-native-community/react-native-video/issues/1319

had to use `react-native-orientation` and 
```js
  onFullscreenPlayerWillPresent = () => {
    Orientation.unlockAllOrientations();
  };

  onFullscreenPlayerWillDismiss = () => {
    Orientation.lockToPortrait();
  };
```

thanks!

**Comment by juancarlosqr** [comment]
@cobarx @jenshandersson @ashnfb @n1ru4l any notion when this could be merged? 🙏🏼

**Comment by gazialankus** [comment]
This works well in iOS12.4, iPhone6. 

However, In iOS 13.6, iPhone 11, this freezes the video when making it fullscreen. When trying to close fullscreen it crashes. This line is the culprit: `_fullscreenPlayerPresented = YES`

**Comment by gazialankus** [comment]
putting it in `dispatch_async(dispatch_get_main_queue(), ^{             });` fixes iPhone 11!

**Comment by gazialankus** [comment]
I sent a PR to this branch that wants to PR here :D https://github.com/24i/react-native-video/pull/15

**Pull request closed without merging**